packages/consumption/src/modules/requests/utility/validateRelationshipAttributesWithinRequest.ts

```diff
@@ -0,0 +1,206 @@
+            if (fragmentsOfMustBeAcceptedItemsOfGroup) {
+                fragmentsOfMustBeAcceptedItemsOfRequest.push(...fragmentsOfMustBeAcceptedItemsOfGroup);
+            }
+        } else {
+            const fragmentOfMustBeAcceptedRequestItem = extractRelationshipAttributeFragmentFromMustBeAcceptedRequestItem(item, recipient);
+            if (fragmentOfMustBeAcceptedRequestItem) {
+                fragmentsOfMustBeAcceptedItemsOfRequest.push(fragmentOfMustBeAcceptedRequestItem);
+            }
+        }
+    }
+
+    const containsMustBeAcceptedDuplicatesResult = containsDuplicateRelationshipAttributeFragments(fragmentsOfMustBeAcceptedItemsOfRequest);
+    if (containsMustBeAcceptedDuplicatesResult.containsDuplicates) {
+        throw ConsumptionCoreErrors.requests.violatedKeyUniquenessOfRelationshipAttributes(
+            `The Request can never be accepted because it would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key '${containsMustBeAcceptedDuplicatesResult.duplicateFragment.key}', owner and value type.`
+        );
+    }
+
+    const fragmentsOfAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+
+    for (let i = 0; i < params.length; i++) {
+        const item = items[i];
+        const decideItemParams = params[i];
+
+        if (item instanceof RequestItemGroup) {
+            const fragmentsOfAcceptedItemsOfGroup = extractRelationshipAttributeFragmentsFromAcceptedItemsOfGroup(item, decideItemParams as DecideRequestItemGroupParametersJSON);
+            if (fragmentsOfAcceptedItemsOfGroup) {
+                fragmentsOfAcceptedItemsOfRequest.push(...fragmentsOfAcceptedItemsOfGroup);
+            }
+        } else {
+            const fragmentOfAcceptedRequestItem = extractRelationshipAttributeFragmentFromAcceptedRequestItem(item, decideItemParams as DecideRequestItemParametersJSON);
+            if (fragmentOfAcceptedRequestItem) {
+                fragmentsOfAcceptedItemsOfRequest.push(fragmentOfAcceptedRequestItem);
+            }
+        }
+    }
+
+    const containsAcceptedDuplicatesResult = containsDuplicateRelationshipAttributeFragments(fragmentsOfAcceptedItemsOfRequest);
+    if (containsAcceptedDuplicatesResult.containsDuplicates) {
+        return ValidationResult.error(
+            ConsumptionCoreErrors.requests.invalidAcceptParameters(
+                `The Request cannot be accepted with this parameters because it would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key '${containsAcceptedDuplicatesResult.duplicateFragment.key}', owner and value type.`
+            )
+        );
+    }
+
+    return ValidationResult.success();
+}
+
+function extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(requestItemGroup: RequestItemGroup, recipient?: CoreAddress): RelationshipAttributeFragment[] | void {
```

Comment:
`void` is a bit more special, there also other instances here

```suggestion
function extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(requestItemGroup: RequestItemGroup, recipient?: CoreAddress): RelationshipAttributeFragment[] | undefined {
```